### DISCUSSION TITLE
USBto Serial

### DISCUSSION THREAD

**Discussion by Pacer27** [Rodemfr/MicronetToNMEA#86] [General]
Hello,
i testet UsbtoSerial (examples Teensy) to configure my Neo-M8N Module with u-center.
It works!

Is it possible to integrate it in the Code?
Maybe with a new Menu option?

**Comment by Rodemfr** [comment]
Hi Pacer27,

In theory, setting `GNSS_UBLOXM8N` to *1* in `BoardConfig.h` should enable M8N initialization code. It does not work for you ?
If not, this might be caused by a faulty connection to the MN8 RX pin (this happened to me).

**Comment by Pacer27** [comment]
Hello,
in my case the Configuration didn't work well. The Neo-M8N has a Firmware 3.51. I have to config a lot of options. By the defaults from you i had have an Derivation of about 25 m.
Therefore it would be nice to be able to use the USB interface of the Tennsy with the U-Center without having to rewrite the module.

Unfortunately I am not a programmer.

**Comment by Rodemfr** [comment]
25m cable might be the problem. The start function in M8NDriver.cpp files configures M8N to 38400 bauds and makes most of the configuration after. 38400 bauds is pretty high for a 25m cable. You can try to comment the two following lines in M8NDriver.cpp :

```
87   GPS_SendConfig(UART1_38400, 28);
88   GNSS_SERIAL.begin(38400);
```

It will let the baudrate to the BoardConfig value (9600 bauds) which will be better for a 25m cable.
